ledger/ledgercore/statedelta.go

```diff
@@ -90,16 +90,19 @@ type AccountDeltas struct {
 	acctsCache map[basics.Address]int
 }
 
-// MakeStateDelta creates a new instance of StateDelta
+// MakeStateDelta creates a new instance of StateDelta.
+// hint is amount of transactions for evaluation, 2 * hint is for sender and receiver balance records.
+// This does not play well for AssetConfig and ApplicationCall transactions on scale
 func MakeStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64, hint int, compactCertNext basics.Round) StateDelta {
 	return StateDelta{
 		Accts: AccountDeltas{
 			accts:      make([]basics.BalanceRecord, 0, hint*2),
 			acctsCache: make(map[basics.Address]int, hint*2),
 		},
-		Txids:                    make(map[transactions.Txid]basics.Round, hint),
-		Txleases:                 make(map[Txlease]basics.Round, hint),
-		Creatables:               make(map[basics.CreatableIndex]ModifiedCreatable, hint),
+		Txids:    make(map[transactions.Txid]basics.Round, hint),
+		Txleases: make(map[Txlease]basics.Round, hint),
+		// asset or application creation are considered as rare events so do not pre-allocate space for them
+		Creatables:               make(map[basics.CreatableIndex]ModifiedCreatable),
```

Comment:
I'm fine with these - make sure you look on OptimizeAllocatedMemory and verify it's good there as well.

Reply:
Does `OptimizeAllocatedMemory` really work? I wonder why/how, if hint is kind of good enough for everything except AssetConfig and ApplicationCall txn.

Reply:
According to Nicholas, it shown a substantial improvement on long running test system.

Reply:
done